src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -1174,6 +1178,58 @@ private bool TryReadLegacyExifTextChunk(ImageMetadata metadata, string data)
             return true;
         }
 
+        /// <summary>
+        /// Reads the color profile chunk. The data is stored similar to the zTXt chunk.
+        /// </summary>
+        /// <param name="metadata">The metadata.</param>
+        /// <param name="data">The bytes containing the profile.</param>
+        private void ReadColorProfileChunk(ImageMetadata metadata, ReadOnlySpan<byte> data)
+        {
+            int zeroIndex = data.IndexOf((byte)0);
+            if (zeroIndex < PngConstants.MinTextKeywordLength || zeroIndex > PngConstants.MaxTextKeywordLength)
+            {
+                return;
+            }
+
+            byte compressionMethod = data[zeroIndex + 1];
+            if (compressionMethod != 0)
+            {
+                // Only compression method 0 is supported (zlib datastream with deflate compression).
+                return;
+            }
+
+            ReadOnlySpan<byte> keywordBytes = data.Slice(0, zeroIndex);
+            if (!this.TryReadTextKeyword(keywordBytes, out string name))
+            {
+                return;
+            }
+
+            ReadOnlySpan<byte> compressedData = data.Slice(zeroIndex + 2);
+
+            using (var memoryStream = new MemoryStream(compressedData.ToArray()))
+            using (var bufferedStream = new BufferedReadStream(this.Configuration, memoryStream))
+            using (var inflateStream = new ZlibInflateStream(bufferedStream))
```

Comment:
I think we can avoid the allocation here. 

https://github.com/dotnet/runtime/issues/24622#issuecomment-356465832